src/wallet/spend.cpp

```diff
@@ -1086,7 +1086,13 @@ util::Result<CreatedTransactionResult> CreateTransaction(
     return res;
 }
 
-bool FundTransaction(CWallet& wallet, CMutableTransaction& tx, CAmount& nFeeRet, int& nChangePosInOut, bilingual_str& error, bool lockUnspents, const std::set<int>& setSubtractFeeFromOutputs, CCoinControl coinControl)
+util::Result<FundedTransactionResult> FundTransaction(
+        CWallet& wallet,
+        CMutableTransaction& tx,
+        int change_pos,
+        bool lockUnspents,
+        const std::set<int>& setSubtractFeeFromOutputs,
+        CCoinControl coinControl)
```

Comment:
since we are here, coin control could be passed as a ref so we avoid copying it.

Reply:
Thanks, done. This is potentially more than just a refactor since the passed in coin control object from the caller is now modified, but as far as I can see this is no problem.